drivers/serial/Kconfig.nrfx

```diff
@@ -198,8 +267,70 @@ config UART_3_NRF_TX_BUFFER_SIZE
 	  This value is limited by range of TXD.MAXCNT register for
 	  particular SoC.
 
+config UART_3_NRF_HW_ASYNC
+	bool "Use hardware RX byte counting"
+	depends on UART_3_ASYNC
+	help
+	  If default driver uses interrupts to count incoming bytes, it is possible
+	  that with higher speeds and/or high cpu load some data can be lost.
+	  It is recommended to use hardware byte counting in such scenarios.
+	  Hardware RX byte counting requires timer instance and one PPI channel
+
+config UART_3_NRF_HW_ASYNC_TIMER
+	int "Timer instance"
+	depends on UART_3_NRF_HW_ASYNC
+
 endif # UART_3_NRF_UARTE
 
+if UART_0_NRF_HW_ASYNC_TIMER = 0 || UART_1_NRF_HW_ASYNC_TIMER = 0 || \
+	UART_3_NRF_HW_ASYNC_TIMER = 0 || UART_2_NRF_HW_ASYNC_TIMER = 0
+config NRFX_TIMER0
+	default y
```

Comment:
Could use a `depends on` on the symbol instead of an `if` here.

`default y if UART_0_HW_ASYNC_TIMER = 0 || ...` will work too.